fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/StoreTimer.java

```diff
@@ -74,6 +76,66 @@ public static void checkEventNameUniqueness(@Nonnull Stream<Event> events) {
         }
     }
 
+    /**
+     * Subtracts counts and times recorded by a snapshot of a timer.
+     *
+     * Subtracting a snapshot of a timer from the original timer after subsequent operations have
+     * been performed and timed can provide useful metrics as to the cost of those subsequent operations.
+     *
+     * The timer must have been previously derived from the snapshot. Moreover,
+     * the timer should not have been reset after the snapshot was taken.
+     *
+     * @param timer timer to subtract the snapshot from
+     * @param timerSnapshot snapshot of the provided timer
+     *
+     * @return a snapshot of the provided timer
+     */
+    @Nonnull
+    public static StoreTimer subtractSnapshot(@Nonnull StoreTimer timer, @Nonnull StoreTimerSnapshot timerSnapshot) {
+
+        if (timer.hashCode() != timerSnapshot.getFromID()) {
```

Comment:
This is OK, however if anyone ever implements `hashCode()` on the timer it could get to be an expensive operation (the default implementation that we have today basically amount to a hash of the memory address of the timer so it is cheap).  This might be another good opportunity to rely on a UUID instead.

Reply:
Will change it to use UUID.